### PR TITLE
[INFRA] test bidsschematools  against osx and windows as well

### PR DESCRIPTION
**Pull request by yarikoptic**
Inspired by https://github.com/bids-standard/bids-specification/issues/1294 and thus should fail one test on windows ATM.

attn @TheChymera . I am ok if fix for #1294 would simply absorb this PR

### CODE REVIEW

File: .github/workflows/schemacode_ci.yml [hunk 1]
```diff
@@ -51,7 +51,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: ["ubuntu-latest"]
+        os: ["ubuntu-latest", "macos-latest", "windows-latest"]
         python-version: ["3.7", "3.8", "3.9", "3.10"]
```
Comment (effigies) [inline .github/workflows/schemacode_ci.yml:55]: Given that it's all filesystem manipulation and not weird numerical edge cases, it seems like overkill to test the whole Python range on all three platforms. I would suggest:

```suggestion
        os: ["ubuntu-latest"]
        python-version: ["3.7", "3.8", "3.9", "3.10"]
        include:
          - os: macos-latest
            python-version: 3
          - os: windows-latest
            python-version: 3
```

**Review by TheChymera — APPROVED** [review]
As per [the other discussion regarding windows](https://github.com/bids-standard/bids-specification/issues/1294) I really think we should consider deprecating this OS entirely and just pointing people to [WSL](https://en.wikipedia.org/wiki/Windows_Subsystem_for_Linux). Windows causes too much trouble with trivial details such as permissions and paths.

*However* given that we seem to have downstream demand, as you mentioned in that discussion, we might be forced to do this for the moment :( What I'd suggest is to keep pushing WSL and revisit this again regularly to avoid having this dangling test constraint in perpetuity. It's a bit strange that we try to push for best data practices, try to get people to use our specific custom standards and software, and don't put an equal or greater amount of effort into getting them to use the far more established standards and software of GNU/linux....

**Comment by yarikoptic** [comment]
ok -- this one is good, but I think we should not merge it since fails on Windows. @TheChymera - please base your fix on top of this PR branch, meanwhile I will just close it (branch should still lurk there)
